tests/python_package_test/test_string.py

```diff
@@ -0,0 +1,102 @@
+# coding: utf-8
+import filecmp
+import numbers
+from pathlib import Path
+
+import numpy as np
+# import pytest
+from scipy import sparse
+from sklearn.datasets import dump_svmlight_file, load_svmlight_file, load_breast_cancer
+from sklearn.model_selection import train_test_split
+import sys
+sys.path.append("python-package")
```

Comment:
Please do not modify `sys.path`, as that could lead to a situation where some imports seem to work during tests and then actually fail when users run `lightgbm` on their systems.